### PR TITLE
Fix provider bug

### PR DESCRIPTION
**Pull request by BeyondEvil**
Fix for issue discussed on Slack (+tests)

### CODE REVIEW

File: pytest_selenium/drivers/browserstack.py [hunk 1]
```diff
@@ -14,13 +14,17 @@ class BrowserStack(Provider):
...
+        return self.username, self.key
 
     @property
     def executor(self):
         return 'http://{0}:{1}@hub.browserstack.com:80/wd/hub'.format(
             self.username, self.key)
 
+    @property
+    def name(self):
+        return type(self).__name__
```
Comment (davehunt) [inline pytest_selenium/drivers/browserstack.py]: We can just inherit this from `Provider`

File: pytest_selenium/drivers/cloud.py [hunk 2]
```diff
@@ -17,15 +17,15 @@ class Provider(object):
 
     @property
     def driver(self):
-        return type(self).__name__
+        return type(self).__name__.lower()
 
     @property
     def name(self):
         return self.driver
```
Comment (davehunt) [inline pytest_selenium/drivers/cloud.py]: Can we switch this around so this returns `type(self).__name__` and `driver` returns `self.name.lower()`?

File: testing/test_driver.py [hunk 3]
```diff
@@ -115,3 +115,37 @@ def test_pass(driver_kwargs):
...
+        import pytest
+        @pytest.mark.nondestructive
+        def test_pass(request):
+            assert request.config.getoption('driver') == 'saucelabs'
+    """)
+    testdir.quick_qa('--driver', 'SauceLabs',
+                     file_test, passed=1)
+
+
+def test_provider_lowercase():
```
Comment (davehunt) [inline testing/test_driver.py]: Let's parameterise this test.

Comment (BeyondEvil) [reply testing/test_driver.py]: I thought about it, but I'm not sure of how to accomplish that with the imports... 🤔

File: pytest_selenium/drivers/browserstack.py (hunk 1 above)
Comment (BeyondEvil) [reply pytest_selenium/drivers/browserstack.py]: Yeah, that's the way it was, but then the tests failed. But if we do your suggestion below and the `uses_driver()` function, we can restore it. 👍

File: pytest_selenium/drivers/cloud.py (hunk 2 above)
Comment (BeyondEvil) [reply pytest_selenium/drivers/cloud.py]: Agreed.

File: testing/test_driver.py (hunk 3 above)
Comment (BeyondEvil) [reply testing/test_driver.py]: yeah, I need help with this @davehunt  (the imports).

File: pytest_selenium/drivers/saucelabs.py [hunk 4]
```diff
@@ -27,6 +27,10 @@ def executor(self):
         return 'https://{0}:{1}@ondemand.saucelabs.com/wd/hub'.format(
             self.username, self.key)
 
+    @property
+    def driver(self):
+        return 'saucelabs'
```
[the remark this answers is not in the document]
Comment (BeyondEvil) [reply pytest_selenium/drivers/saucelabs.py]: We can (incl. the entire `driver` property), but to do that, one of two other things must change.

1. "Sauce Labs" needs to become "SauceLabs". That is, removing the overloaded `name` property in the `SauceLabs` class.

2. `uses_driver` needs to be overloaded to remove the space.

```
    def uses_driver(self, driver):
        return driver.lower() == self.name.lower().replace(' ', '')
```

File: testing/test_driver.py [hunk 5]
```diff
@@ -115,3 +118,14 @@ def test_pass(driver_kwargs):
...
+
+@pytest.mark.parametrize(('module_name', 'class_name'),
+                         [(saucelabs, 'Sauce Labs'),
+                          (testingbot, 'TestingBot'),
+                          (crossbrowsertesting, 'CrossBrowserTesting'),
+                          (browserstack, 'BrowserStack')])
+def test_provider_naming(module_name, class_name):
+    provider = getattr(module_name, class_name.replace(' ', ''))()
+    assert provider.uses_driver(class_name.replace(' ', ''))
+    assert provider.name == class_name
```
[the remark this answers is not in the document]
Comment (BeyondEvil) [reply testing/test_driver.py]: Ah, nice!

File: pytest_selenium/drivers/cloud.py [hunk 6]
```diff
@@ -41,3 +42,6 @@ def get_credential(self, key, envs):
                 if value:
                     return value
         raise MissingCloudCredentialError(self.name, key, envs)
+
+    def uses_driver(self, driver):
+        return driver.lower() == self.driver
```
[the remark this answers is not in the document]
Comment (BeyondEvil) [reply pytest_selenium/drivers/cloud.py]: Yes, afaict the `driver` property is only used in one other place, and that is `config()` (for getting the dot-configuration files), and that can use the `name` property just as well.

File: pytest_selenium/drivers/saucelabs.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (BeyondEvil) [reply pytest_selenium/drivers/saucelabs.py]: Let's go with option 1 @davehunt

**Pull request merged**